File: sigedu-frontend-final/src/api/docenteService.js
import axios from './axios';

export const docenteService = {
    getAll: async () => {
        const response = await axios.get('/docentes');
        return response.data;
    },
    getById: async (id) => {
        const response = await axios.get(`/docentes/${id}`);
        return response.data;
    },
    getByDni: async (dni) => {
        const response = await axios.get(`/docentes/dni/${dni}`);
        return response.data;
    },
    getByCodigo: async (codigo) => {
        const response = await axios.get(`/docentes/codigo/${codigo}`);
        return response.data;
    },
    getByEspecialidad: async (especialidad) => {
        const response = await axios.get(`/docentes/especialidad/${especialidad}`);
        return response.data;
    },
    create: async (docente) => {
        const response = await axios.post('/docentes', docente);
        return response.data;
    },
    update: async (id, docente) => {
        const response = await axios.put(`/docentes/${id}`, docente);
        return response.data;
    },
    delete: async (id) => {
        const response = await axios.delete(`/docentes/${id}`);
        return response.data;
    },
    reactivar: async (id) => {
        const response = await axios.patch(`/docentes/${id}/reactivar`);
        return response.data;
    },
    getActivos: async () => {
        const response = await axios.get('/docentes/activos');
        return response.data;
    }
};

File: sigedu-backend-final/src/main/java/com/sigedu/backend/service/DocenteService.java
package com.sigedu.backend.service;

import com.sigedu.backend.entity.Docente;
import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.Optional;

public interface DocenteService {
    List<Docente> findAll();
    Page<Docente> findAll(Pageable pageable);
    List<Docente> findAllActivos();  // NUEVO
    Optional<Docente> findById(Integer id);
    Optional<Docente> findByDni(String dni);
    Optional<Docente> findByCodigo(String codigo);
    Docente save(Docente docente);
    Docente update(Integer id, Docente docente);
    void delete(Integer id);
    void reactivar(Integer id);  // NUEVO
    List<Docente> findByEspecialidad(String especialidad);
    Page<Docente> buscarPaginado(String termino, Boolean estado, Pageable pageable);
}